### PR TITLE
Refactor: Extract getStartUrl hook from site migration

### PR DESCRIPTION
**Pull request by andres-blanco**
Related to #

## Proposed Changes

As I was making experimentation for the site-hosted-migration flow I found this getStartUrl method in the way. It was making the code harder to read.

## Testing Instructions

Go through the site-migration flow exercising this hook. To do this go to `/setup/site-migration` while logged out and see that you are redirected. 

Also check the test passes.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/landing/stepper/hooks/use-start-url.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { useLocale } from '@automattic/i18n-utils';
+import { getLocaleFromPathname, getLocaleFromQueryParam } from 'calypso/boot/locale';
+import { getLoginUrl } from '../utils/path';
+
+export const useStartUrl = ( flowName: string ) => {
+	const useLocaleSlug = useLocale();
+	// Query param support can be removed after dotcom-forge/issues/2960 and 2961 are closed.
+	const queryLocaleSlug = getLocaleFromQueryParam();
```
Comment (gabrielcaires) [inline client/landing/stepper/hooks/use-start-url.ts:8]: Can we already remove this code?

File: client/landing/stepper/hooks/use-start-url.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { useLocale } from '@automattic/i18n-utils';
+import { getLocaleFromPathname, getLocaleFromQueryParam } from 'calypso/boot/locale';
+import { getLoginUrl } from '../utils/path';
```
Comment (daledupreez) [inline client/landing/stepper/hooks/use-start-url.ts]: While looking at a bunch of this code over the last few days, I found two really helpful hooks that I think we should use in this code.
```suggestion
import { useFlowLocale } from 'calypso/landing/stepper/hooks/use-flow-locale';
import { useLoginUrl } from '../utils/path';
```

Maybe it's worth tackling that refactor in a follow-up, but I wanted to flag it regardless.

Comment (andres-blanco) [reply client/landing/stepper/hooks/use-start-url.ts]: Nice, I've applied the change!

**Review by sixhours — APPROVED** [review]
Nice clean up! Working for me 👍

**Pull request merged**